Live/Classes/GameObject.cpp

```diff
@@ -2,8 +2,13 @@
 
 USING_NS_CC;
 
-GameObject::GameObject(const std::string sprite_file, unsigned int grid_pos_x,
-                       unsigned int grid_pos_y)
+GameObject::GameObject(const std::string sprite_file) : m_sprite_file(sprite_file) {
```

Comment:
Use default values instead, i.e. in the header, do `GameObject(const std::string sprite_file, int grid_pos_x = -1, int grid_pos_y = -1)` and you wouldn't need this ctor at all.